src/components/displayDocuments/documents.tsx

```diff
@@ -6,22 +6,35 @@ export interface Document {
   version: "2" | "3";
 }
 
+enum ChainId {
+  Local = 1337,
+  Ethereum = 1,
+  Ropsten = 3,
+  Rinkeby = 4,
+  Goerli = 5,
+  Polygon = 137,
+  PolygonMumbai = 80001,
+  Sepolia = 11155111
+}
+
 interface Action {
   uri: string;
   key?: string;
   permittedActions: string[];
   redirect: string;
+  chainId?: ChainId;
 }
 
 const stringifyAndEncode = (obj: object): string => window.encodeURIComponent(JSON.stringify(obj));
 
-const uriToAction = ({ uri, key, permittedActions, redirect }: Action): string => {
+const uriToAction = ({ uri, key, permittedActions, redirect, chainId }: Action): string => {
   const action = stringifyAndEncode({
     type: "DOCUMENT",
     payload: {
       uri,
       permittedActions,
-      redirect
+      redirect,
+      chainId
```

Comment:
umm... Do you happen to know where will the `chainId` you've added into this payload be used in the codebase?

Reply:
[here](https://github.com/TradeTrust/tradetrust-website/pull/607/files#diff-50f4a09a0dbd70aaa0c8c5e3d78198ca60a5b3898196d92f298da527e0c6f6c9R48)

Reply:
umm... That link goes to the tradetrust website repo, but I don't see this repo being imported into the website. Did I miss something?🙈

Reply:
Thanks for going through with me how the `chainId` is used just now!😀 The only tricky thing with this is the `chainId` from the action takes precedence over the one in the network object in the file. Nevertheless, we have concluded that the gallery is managed by us and so the `chainId`s in the actions are also supplied by us on the page. Hence, the `chainId` coming from the action is expected to be as intended.